### PR TITLE
CVE and path lines scripts

### PR DESCRIPTION
**Pull request by justyoyo123**
All the scripts that are needed to get all the CVEs associated with a given kernel version, put all pop path lines into a dictionary, and then compare the CVE line number with the pop paths data.

### CODE REVIEW

File: cve_scripts/cve_scraper.py
Comment (rennergade) [inline cve_scripts/cve_scraper.py]: We're going to want to read this from a config file instead of user input.

File: cve_scripts/cve_scraper.py
Comment (rennergade) [inline cve_scripts/cve_scraper.py:4]: Just want to confirm selenium should work headless right? and on linux?

File: cve_scripts/paths2json.py
Comment (rennergade) [inline cve_scripts/paths2json.py:14]: did we not find a better way to deal with this instead of this massive elif chain?

File: cve_scripts/pop_paths_comparison.py
Comment (rennergade) [inline cve_scripts/pop_paths_comparison.py:36]: again

File: cve_scripts/pop_paths_comparison.py
Comment (rennergade) [inline cve_scripts/pop_paths_comparison.py]: again

**Review by rennergade — COMMENTED** [review]
Just did a quick pass. Made some comments. Will do another one once the other guys look over it.

File: cve_scripts/cve_scraper.py
Comment (TristanB22) [inline cve_scripts/cve_scraper.py:1]: Just a general note -- this entire script could maybe do with a readme to help people understand what to change if things with the website change in the future. For example, what if the website structure changes and the person looking at this doesn't understand how to simply fix the script so that it can still run on the new folder structure?

File: cve_scripts/paths2json.py
Comment (TristanB22) [reply cve_scripts/paths2json.py:14]: try a switch statement maybe? or a for loop with an array of the options which breaks once one of the options is met?

File: cve_scripts/pop_paths_comparison.py
Comment (TristanB22) [inline cve_scripts/pop_paths_comparison.py:4]: These comments are kinda confusing me -- is this something that the user is supposed to do or something that you are reminding yourself to do later? If it is for the user, then we definitely need a readme going here.

File: cve_scripts/run.sh
Comment (TristanB22) [inline cve_scripts/run.sh]: Overall looks really good -- good job!

**Review by TristanB22 — COMMENTED** [review]
Awesome -- once you address Nick and I's comments this looks like it should be ready to go. I assume that you have already tested the script so I didn't spend an insane amount of time looking at the actual logic behind the code. The biggest thing was the readme.

File: cve_scripts/cve_scraper.py
Comment (justyoyo123) [reply cve_scripts/cve_scraper.py:4]: Yes selenium works headless. Yes it runs on linux, the user running the script needs to add the path to the chrome binary file on their laptop where I commented. I will put that in the readme that Tristan suggested making.

File: cve_scripts/paths2json.py
Comment (justyoyo123) [reply cve_scripts/paths2json.py:14]: Okay, elif chain has been removed in latest commit with a for loop over an array of the file path prefixes

File: cve_scripts/cve_scraper.py
Comment (justyoyo123) [reply cve_scripts/cve_scraper.py:1]: Okay, will add that to readme

File: cve_scripts/cve_scraper.py
Comment (rennergade) [reply cve_scripts/cve_scraper.py:4]: Ok we should make that part of the config

File: cve_scripts/paths2json.py
Comment (rennergade) [inline cve_scripts/paths2json.py:4]: im confused where this pathname is coming from?

File: config.sh
Comment (rennergade) [inline config.sh]: i dont think we need to let the user decide where the patch files are, there sould just be a folder were all intermediate files are

File: config.sh
Comment (rennergade) [inline config.sh:16]: you changed these to be added to the config files, but didnt change Tristan's scripts to use ENV variables

**Review by rennergade — CHANGES_REQUESTED** [review]

File: cve_scripts/paths2json.py
Comment (justyoyo123) [reply cve_scripts/paths2json.py:4]: For this, I looked through Tristan's scripts to look for the output file of the pop paths data which I think happens in analyzer.py but I don't see an explicit output file.

Comment (rennergade) [reply cve_scripts/paths2json.py:4]: ok yeah we need to change this to where tristans pop path file is outputted. @TristanB22 where is that exactly?

File: pop_paths_scripts/install_kernel_trinity_ltp.sh
Comment (rennergade) [inline pop_paths_scripts/install_kernel_trinity_ltp.sh]: im pretty sure this isnt the correct path since this is in the pop_paths_scripts folder. I also think you missed a few other files where tristan was using the .txt config style

Comment (justyoyo123) [reply pop_paths_scripts/install_kernel_trinity_ltp.sh]: Yeah my bad, forgot to check host scripts. I think I got them all now.

File: cve_scripts/paths2json.py
Comment (TristanB22) [reply cve_scripts/paths2json.py:4]: in host_scripts/ssh_command.sh -- " sudo scp -r -i $(cat ../directory.txt)/key -o StrictHostKeyChecking=no -P 2222 ubuntu@localhost:/home/ubuntu/GCOV_DATA $(cat ../directory.txt)/GCOV_DATA/$(date --iso-8601="minutes")"

When I was working on this script this command would sometimes block -- I was never able to figure out why exactly.

Comment (TristanB22) [reply cve_scripts/paths2json.py:4]: Lol that formatted super strangely because of markdown but I think you know what I mean... /pop-paths-scripts/host_scripts/ssh_command.sh line 58

Comment (rennergade) [reply cve_scripts/paths2json.py:4]: @TristanB22 does your script generate an analyzer file? Thats what were looking for

Comment (TristanB22) [reply cve_scripts/paths2json.py:4]: No, but that shouldn't be too difficult to create since we know the path to the data and just need to feed that to the analyzer. Was doing it manually before because we were just looking to get the data files instead of the analyzer files before.

**Review by TristanB22 — APPROVED** [review]
I think that this is starting to look a lot better. No major issues from my end.

File: cve_scripts/cve_scraper.py
Comment (rennergade) [reply cve_scripts/cve_scraper.py:4]: @justyoyo123 did you end up adding this to config?

File: cve_scripts/paths2json.py
Comment (rennergade) [reply cve_scripts/paths2json.py:4]: ok @TristanB22 will add this after we merge this PR

**Review by rennergade — APPROVED** [review]
I think this is ready to merge if the selenium binary path thing is fixed.

**Review by rennergade — APPROVED** [review]
Ok this is good.

**Pull request merged**
